src/main/java/org/jenkinsci/plugins/codesonar/services/CodeSonarCacheService.java

```diff
@@ -81,13 +95,20 @@ private AlertsService getAlertsService() throws CodeSonarPluginException {
         return alertsService;
     }
     
+    private WarningsService getWarningsService() throws CodeSonarPluginException {
+        if (warningsService == null) {
+            warningsService = new WarningsService(httpService, hubInfo.isStrictQueryParametersEnforced());
+        }
+        return warningsService;
+    }
+    
     private CodeSonarAnalysisData getAnalysisDataFromCache(URI baseHubUri, long analysisId) {
         LOGGER.log(Level.INFO, "Getting analysis from cache: baseHubUri = {0}, analysisId = {1,number,0}", new Object[] {baseHubUri, analysisId});
         Pair<URI, Long> key = Pair.with(baseHubUri, analysisId);
         CodeSonarAnalysisData analysisData = cachedAnalyses.get(key);
         if(analysisData == null) {
             LOGGER.log(Level.INFO, "Analysis data not found from cache, putting new instance into cache");
-            analysisData = new CodeSonarAnalysisData();
+            analysisData = new CodeSonarAnalysisData(baseHubUri, analysisId);

```

Comment:
It's useful to keep both `baseHubUri` and `analysisId` in this class for subsequent requests to the hub (i.e. on demand data retrieval).